sympy/polys/polymatrix.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import print_function
+
+from sympy.matrices.dense import MutableDenseMatrix
+from sympy.polys.polytools import Poly
+from sympy.core.compatibility import reduce
```

Comment:
I think it wouldn't be a problem to use the `reduce` from SymPy instead of python's `reduce`.